### ISSUE TITLE
dplyr::mutate and database back ends

### ISSUE THREAD

**Issue by kylelundstedt** [data-cleaning/dcmodify#9]
Hi @markvanderloo 

I love what you and @edwindj are doing with the validate, error locate, and dcmodify packages!  I'm looking forward to putting them to use on our reams of mortgage data.  I cannot agree more that data cleaning rules and data modification rules should be stored in separate, version controlled plain text files.  Your approach is critical for replicable production workflows.

That said, most of our production workflows rely upon using the 'dplyr' package in R to generate the SQL necessary to work with our significant volumes (multiple TB) of data, either stored in Amazon Redshift or processed using Spark.  

Is there a way for your data cleaning packages to work with dplyr, so that we get the benefit of simple version-controlled text-based rule files, yet the scalability of 'dplyr'-based SQL?  I'd appreciate any insight you could offer.

Cheers
Kyle

**Comment by markvanderloo** [comment]
Thanks for the support!

At the moment we do not suport `tbl_df` objects yet, but we do intend it. The packages are set up using S4 & reference classes to support both overloading on data and rule types. 
Basically, our objects storing rules are just lists of `call` objects with some metadata. Converting them to expressions and then releasing `dplyr::translate_sql` on them would go a long way of supporting such actions.

The bottom line is tha: our time is limited so its not done yet, even though the supporting technology is there (in e.g. dplyr). We would consider PR's that implement such functionality, starting with `validate`. Other option would be to have a separate package (`validate.sql` for example) that implements such backends...

**Comment by edwindj** [comment]
Working on that: https://github.com/data-cleaning/dcmodifydb
Feedback is more then welcome
